code-challenges.test.js

```diff
@@ -16,52 +16,163 @@
+        expect(oddOrEven(object3)).toEqual("Not divisible by three")
+    })
+})
+// FAIL  ./code-challenges.test.js
+//   exampleFunction
+    // ✕ takes a object as an argument and decides if the number inside it is evenly divisible by three or not. (1 ms)
+
+    // ● oddOrEven › takes a object as an argument and decides if the number inside it is evenly divisible by three or not.
+
+    // ReferenceError: oddOrEven is not defined
 
 // b) Create the function that makes the test pass.
 
+const oddOrEven = (object) => {
+    const num = object.number;
+    if (num % 3 === 0) {
+        return "Divisible by three";
+    } else {
+        return "Not divisible by three";
+    }
+}
+
 // --------------------2) Create a function that takes in an array of words and returns an array with all the words capitalized.
 
 // Pseudo code:
+// input: Array of words
+// output: array of words with all letters capitalized 
+// function name: toCaps
+// use array of words as an argument for function toCaps
+// using a .map to itereate through each array and index
+// return an array with all words capitalized using .toUpperCase
 
 // a) Create a test with expect statements for each of the variables provided.
+describe("toCaps", () => {
+    it("function that takes in an array of words and returns an array with all the words capitalized.", () => {
+        const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
+        // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
+        const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
+        // Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
+        expect(toCaps(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+        expect(toCaps(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango"])
+    })
+})
+
+// FAIL  ./code-challenges.test.js
+//   toCaps
+//     ✕ function that takes in an array of words and returns an array with all the words capitalized. (3 ms)
 
-const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
-// Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
-const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
-// Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
+//   ● toCaps › function that takes in an array of words and returns an array with all the words capitalized.
+
+//     ReferenceError: toCaps is not defined
 
 // b) Create the function that makes the test pass.
 
+const toCaps = (array) => {
+    return array.map(string => string.charAt(0).toUpperCase() + string.slice(1));
+  }
+
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
 
 // Pseudo code:
+// input: string
+// output: string index of first vowel 
+// function name: firstVowel
+// create an array of vowels
+// iterate through the array with a for loop 
```

Comment:
Your for loop is not set to run for the length of the array, but for the length of the inputted string, so it would be iterating through the string and checking if each character in the string is included in the vowel array you created. That being said, I am loving your pseudo code and how you are breaking each step into clear and concise descriptions.